### PR TITLE
Optimize const executions

### PR DESCRIPTION
**Pull request by shargon**
Optimized `NOTEQUAL`,  `SIGN`,`ABS`,  `NEGATE`,`INC`, `DEC`,`ADD`,`SUB`,`MUL`, `DIV`, `MOD`,`ISNULL`

### CODE REVIEW

File: src/Neo.Compiler.MSIL/Optimizer/Parser_DeleteStaticMath.cs [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+                                items[x - 2] is NefInstruction p2 &&
+                                p1.IsPush(out var v1) &&
+                                p2.IsPush(out var v2)
+                                )
+                            {
+                                items.RemoveRange(x - 2, 2);
+                                OptimizedCount++;
+                                x -= 2;
+
+                                ins.UpdateForPush(v2 / v1);
```
Comment (Tommo-L) [inline src/Neo.Compiler.MSIL/Optimizer/Parser_DeleteStaticMath.cs:163]: Should we check the range of value, like v1 is zero.

File: src/Neo.Compiler.MSIL/Optimizer/NefOptimizeTool.cs [hunk 2]
```diff
@@ -13,19 +14,16 @@ public static class NefOptimizeTool
...
-            }
-            , out _);
+            return Optimize(script, out _);
         }
 
         public static byte[] Optimize(byte[] script, out Dictionary<int, int> addrConvertTable)
         {
             return Optimize(script, new OptimizeParserType[]
             {
+                //OptimizeParserType.DELETE_DEAD_CODDE,
```
Comment (shargon) [inline src/Neo.Compiler.MSIL/Optimizer/NefOptimizeTool.cs:24]: https://github.com/neo-project/neo-devpack-dotnet/pull/219/files added a bug that make this optimization fail

**Comment by shargon** [comment]
Why https://github.com/neo-project/neo-devpack-dotnet/pull/252/commits/487b1582a94f96d4a8bb77a503f1bc3333c45b0f decrease the coverage in 15%? @neo-project/ngd-shanghai

**Comment by shargon** [comment]
@neo-project/ngd-shanghai after abi changes seems that coverage it's wrong calculated sometimes.

Take a look to these changes, they decrease the coverage 15%

https://github.com/neo-project/neo-devpack-dotnet/pull/252/commits/487b1582a94f96d4a8bb77a503f1bc3333c45b0f

https://github.com/neo-project/neo-devpack-dotnet/pull/252/commits/527b19e2fdf81a881c949a225c60ef14b963ffbc

**Comment by shargon** [comment]
@lightszero could be merged?

**Pull request merged**